### PR TITLE
Fix EditableText.enableInteractiveSelection on desktop & web

### LINKED ISSUE #48099: enableInteractiveSelection:false for TextFormField not working in Flutter Web (tried Chrome)
This works in Flutter mobile(stable).

### PR DESCRIPTION
**Pull request by jpnurmi**
## Description

`selectionEnabled` was not taken into account in:
- key movement handling in `RenderEditable`
- mouse drag handling in `TextSelectionGestureDetectorBuilder`

## Related Issues

Fixes: #48099

## Tests

I added the following tests:

- rendering/editable_test.dart: 'respects enableInteractiveSelection'
- widgets/text_selection_test.dart: 'test TextSelectionGestureDetectorBuilder mouse drag disabled'

## Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/widgets/text_selection.dart [hunk 1]
```diff
@@ -1077,15 +1077,16 @@ class TextSelectionGestureDetectorBuilder {
...
-    final PointerDeviceKind? kind = details.kind;
-    _shouldShowSelectionToolbar = kind == null
-      || kind == PointerDeviceKind.touch
-      || kind == PointerDeviceKind.stylus;
-
-    renderEditable.selectPositionAt(
-      from: details.globalPosition,
-      cause: SelectionChangedCause.drag,
-    );
+    if (delegate.selectionEnabled) {
```
Comment (jpnurmi) [inline packages/flutter/lib/src/widgets/text_selection.dart:1080]: These could return early if a smaller diff is preferred. All the other input handlers around had it like this, so I chose to keep it consistent with the surroundings.

Comment (gspencergoog) [reply packages/flutter/lib/src/widgets/text_selection.dart:1080]: Actually, returning early is preferred, in both cases.

**Review by gspencergoog — CHANGES_REQUESTED** [review]
Looks pretty good, just switch to early returns, and I think this is good to go.

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Pull request merged**